autorest/preparer.go

```diff
@@ -183,6 +183,16 @@ func WithBaseURL(baseURL string) PrepareDecorator {
 	}
 }
 
+// WithCustomBaseURL returns a PrepareDecorator that replaces brace-enclosed keys within the
+// request base URL (i.e., http.Request.URL) with the corresponding values from the passed map.
+func WithCustomBaseURL(baseURL string, urlParameters map[string]interface{}) PrepareDecorator {
```

Comment:
Hmmm, I would support the above proposal to move them all to `map[string]string`. Even better would be `map[string]fmt.Stringer` [Stringer Type documentation](https://golang.org/pkg/fmt/#Stringer)